empress/recon_vis/recon_viewer.py

```diff
@@ -96,83 +234,387 @@ def render_parasite_helper(fig, node, recon, host_lookup, show_internal_labels,
+
+
+def _event_color_shape(event: recon.Event):
+    """
+    Gives the color and shape for drawing event, depending on event type
+    :param event: Event object
+    :return A tuple with the color and shape of an event
+    """
+    if event.event_type is recon.EventType.TIPTIP:
+        return render_settings.LEAF_NODE_COLOR, render_settings.LEAF_NODE_SHAPE
+    if event.event_type is recon.EventType.COSPECIATION:
+        return render_settings.COSPECIATION_NODE_COLOR, render_settings.COSPECIATION_NODE_SHAPE
+    if event.event_type is recon.EventType.DUPLICATION:
+        return render_settings.DUPLICATION_NODE_COLOR, render_settings.DUPLICATION_NODE_SHAPE
+    if event.event_type is recon.EventType.TRANSFER:
+        return render_settings.TRANSFER_NODE_COLOR, render_settings.TRANSFER_NODE_SHAPE
+    return None, None 
+
+
+def _set_host_node_layout(host_tree: tree.Tree):
     """
     Sets the logicalRow and logicalCol values of each Node in host_tree.
     Assumes that each host Node has its order set already and this function
     uses those values and structure of the tree to set the logicalRow and logicalCol
     :param host_tree:  A Tree object representing the host tree
     :return None
     """
-    # sets logical row values for leaves in the order they appear in the list of host tree leaves
+    # Sets logical row values for leaves in the order they appear in the list of host tree leaves
     logical_row_counter = 0
     for leaf in host_tree.leaf_list():
         leaf.layout.row = logical_row_counter
-        leaf.layout.x = leaf.layout.col  # This can be scaled if desired
-        leaf.layout.y = leaf.layout.row  # This can be scaled if desired
+        leaf.layout.x=leaf.layout.col           # This can be scaled if desired
+        leaf.layout.y=leaf.layout.row           # This can be scaled if desired
         logical_row_counter += 1
-    # helper function to assign row values, postorder traversal
-    set_internal_host_nodes(host_tree.root_node)
+    # Helper function to assign row values, postorder traversal
+    _set_internal_host_nodes(host_tree.root_node)
 
 
-def set_internal_host_nodes(node: tree.Node):
-    """ Helper function for set_host_node_layout. """
-    if node.is_leaf():
+def _set_internal_host_nodes(node: tree.Node):
+    """
+    Helper function for set_host_node_layout
+    :param node: Host Node object that will be rendered
+    """
+    if node.is_leaf:
         return
-    set_internal_host_nodes(node.left_node)
-    set_internal_host_nodes(node.right_node)
+    _set_internal_host_nodes(node.left_node)
+    _set_internal_host_nodes(node.right_node)
     node.layout.row = (node.left_node.layout.row + node.right_node.layout.row) / 2
-    node.layout.x = node.layout.col  # This can be scaled if desired
-    node.layout.y = node.layout.row  # This can be scaled if desired
+    node.layout.x=node.layout.col         # This can be scaled if desired
```

Comment:
Please add the space between `=` back and use two spaces before `#`

Reply:
Fixed